scripts/remote-heal/remote-healer.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/python
+# vim: expandtab:tabstop=4:shiftwidth=4
+'''
+    Tool to process and take action on incoming zabbix triggers.
+'''
+# Disabling invalid-name because pylint doesn't like the naming conention we have.
+# pylint: disable=invalid-name
+
+import argparse
+import ConfigParser
+import logging
+import os
+import re
+import shlex
+import subprocess
+import sys
+
+class RemoteHealer(object):
+    ''' Class to process zabbix triggers and take appropriate actions '''
+    def __init__(self):
+        self.CONFIG_FILE = '/etc/openshift_tools/remote_healer.conf'
+        # Default creds loader
+        self._creds_prefix = '/usr/local/bin/autokeys_loader'
+        self.load_config()
+
+        self._args = self.parse_args()
+
+        self.setup_logging()
+        logging.debug("Got args: " + str(self._args))
+
+    @staticmethod
+    def run_cmd(cmd):
+        ''' Run passed in command (list-separated arguments) '''
+        logging.debug("running: %s", ' '.join(cmd))
+
+        try:
+            subprocess.call(cmd)
+        except OSError:
+            logging.info("failed to run: %s", ' '.join(cmd))
+
+    def cmd_builder(self, cmd):
+        ''' Build command with default or user-provided prefix '''
+        new_cmd = [self._creds_prefix]
+        new_cmd.extend(cmd)
+        return new_cmd
+
+    def ossh_cmd(self, host, cmd):
+        ''' Build command using ossh as root to specified host '''
+        ssh_cmd = ['ossh', host, '-l', 'root', '-c', cmd]
+        return self.cmd_builder(ssh_cmd)
+
+
+    @staticmethod
+    def parse_args():
+        ''' Parse command line arguments passed in through the
+            SSH_ORIGINAL_COMMAND environment variable.
+            Also handle when run manually. '''
+        my_args = None
+
+        parser = argparse.ArgumentParser(description='Take trigger values ' +
+                                         'from command line or ' +
+                                         'SSH_ORIGINAL_COMMAND and take ' +
+                                         'appropriate healing actions')
+        parser.add_argument("--host", required=True)
+        parser.add_argument("--trigger", required=True)
+        parser.add_argument("--trigger-val", required=True)
+        parser.add_argument("--verbose", action="store_true", help='Print to stdout')
+        parser.add_argument("--debug", action="store_true", help='Log more details')
+
+        if os.getenv("SSH_ORIGINAL_COMMAND"):
```

Comment:
I'm not a big fan of this approach since I can't do "ssh root@tower 'do-this --host foo; do-that --host bar; remote-healer --host baz". Since SSH_ORIGINAL_COMMAND would be set, it would go down the env var parser code path, and it would surprise me when I had my heal action run on host foo instead of host baz.  I prefer to have the force command use an explicit option that tells the script to look at the SSH_ORIGINAL_COMMAND, and have it otherwise use normal argv parsing. That being said, it's not really a security thing, and my example is pretty contrived. I just don't think it would be much fun to figure out why things weren't working in a situation like that.